packages/mui-material/src/TextareaAutosize/TextareaAutosize.tsx

```diff
@@ -62,16 +65,21 @@ const TextareaAutosize = React.forwardRef(function TextareaAutosize(
   const { onChange, maxRows, minRows = 1, style, value, ...other } = props;
 
   const { current: isControlled } = React.useRef(value != null);
-  const inputRef = React.useRef<HTMLTextAreaElement>(null);
-  const handleRef = useForkRef(forwardedRef, inputRef);
-  const heightRef = React.useRef<number>(null);
-  const shadowRef = React.useRef<HTMLTextAreaElement>(null);
+  const textareaRef = React.useRef<HTMLTextAreaElement>(null);
+  const handleRef = useForkRef(forwardedRef, textareaRef);
+  const heightRef = React.useRef<number | null>(null);
```

Comment:
Nit: adding `| null` is not necessary: 

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/30d219e45dba049983c1f1d236fd046279b18b53/types/react/index.d.ts#L1723-L1734